selfcal_helpers.py

```diff
@@ -1187,19 +1187,16 @@ def get_SNR_self_individual(vislist,selfcal_library,n_ant,solints,integration_ti
          solint_snr_per_spw[solint]={}       
          if solint == 'inf_EB':
             SNR_self_EB=np.zeros(len(selfcal_library['vislist']))
-            SNR_self_EB_spw=np.zeros([len(selfcal_library['vislist']),len(selfcal_library[maxspwvis]['spwsarray'])])
-            SNR_self_EB_spw_mean=np.zeros([len(selfcal_library[maxspwvis]['spwsarray'])])
             SNR_self_EB_spw={}
             for i in range(len(selfcal_library['vislist'])):
                SNR_self_EB[i]=SNR/((n_ant)**0.5*(selfcal_library['Total_TOS']/selfcal_library[selfcal_library['vislist'][i]]['TOS'])**0.5)
                SNR_self_EB_spw[selfcal_library['vislist'][i]]={}
                for spw in selfcal_library[selfcal_library['vislist'][i]]['spwsarray']:
-                  if spw in SNR_self_EB_spw[selfcal_library['vislist'][i]].keys():
```

Comment:
I don't think this if statement should be removed. This will cause issues, particularly VLA data, where you might have different numbers of spws for executions of the same SB because some baseline boards were unavailable and there were less than 64 spws possible.

Reply:
Ah, good catch - I'm not actually sure why I took this out in the first place, but will put it back in and run some tests.

Reply:
Ok, I took a closer look at this, and the reason I removed the if statement is because a few lines above SNR_self_EB_spw is defined to be {}. So the if statement can never evaluate to true because the line to populate spw in SNR_self_EB_spw is inside the if statement.

I *think* that the for loop above the if statement, though, should handle spw lists that vary between EBs? Because selfcal_library[vis]['spwsarray'] should only contain the spws in vis.

That said... now looking at the lines below, there might be issues when spws change between EBs. Will mark those lines with a separate comment in a moment in case we want to discuss this particular line further here.